scripts/ci/build_task_generator.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+
+from apt import Cache
+
+from ros_buildfarm.argument import \
+    add_argument_distribution_repository_key_files
+from ros_buildfarm.argument import add_argument_build_tool
+from ros_buildfarm.argument import add_argument_distribution_repository_urls
+from ros_buildfarm.argument import add_argument_dockerfile_dir
+from ros_buildfarm.argument import add_argument_env_vars
+from ros_buildfarm.argument import add_argument_ros_version
+from ros_buildfarm.argument import check_len_action
+from ros_buildfarm.common import get_binary_package_versions
+from ros_buildfarm.common import get_distribution_repository_keys
+from ros_buildfarm.common import get_user_id
+from ros_buildfarm.templates import create_dockerfile
+
+
+def main(argv=sys.argv[1:]):
+    parser = argparse.ArgumentParser(
+        description="Generate a 'Dockerfile' for the CI job")
+    parser.add_argument(
```

Comment:
For a lot of the command line arguments the Python module `ros_buildfarm/argument` contains functions to add them. Please use them were possible and when declaring new arguments more than once move them into that module.

Reply:
I surveyed all of the new scripts and could only find one spot where I could switch to one of the provided argument functions.

The biggest hold-up here is that the provided functions add os/arch/etc arguments as positional, where the pattern is to use the option-style in most places. Even though it means duplication, I think that deviating from the pattern used in other scripts with the same purpose is less desirable.

Reply:
> The biggest hold-up here is that the provided functions add os/arch/etc arguments as positional, where the pattern is to use the option-style in most places.

If the argument module only provides functions to add positional arguments (and they are used throughout the code - e.g. `add_argument_rosdistro_name()` 47 times) how can the option-style be the case used in most places? So I would argue that these new files should use the same style as the existing one which seem to be positional arguments for the required ones.